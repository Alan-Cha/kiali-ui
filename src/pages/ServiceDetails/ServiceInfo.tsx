import * as React from 'react';
import ServiceId from '../../types/ServiceId';
import { ServiceInfoDeployments, ServiceInfoRules, ServiceInfoRoutes, ServiceInfoDescription } from './ServiceInfo/';
import { Endpoints, Deployment, Port, Rule } from '../../types/ServiceInfo';
import * as API from '../../services/Api';
import { ToastNotification, ToastNotificationList, Col, Row } from 'patternfly-react';

type ServiceInfoState = {
  labels?: Map<string, string>;
  type: string;
  name: string;
  ip: string;
  ports?: Port[];
  endpoints?: Endpoints[];
  deployments?: Deployment[];
  rules?: Rule[];
  dependencies?: Map<string, string[]>;
  error: boolean;
  errorMessage: string;
};

class ServiceInfo extends React.Component<ServiceId, ServiceInfoState> {
  constructor(props: ServiceId) {
    super(props);
    this.state = {
      labels: new Map(),
      name: '',
      type: '',
      ip: '',
      ports: [],
      deployments: [],
      rules: [],
      dependencies: new Map(),
      error: false,
      errorMessage: ''
    };
  }

  componentWillMount() {
    this.fetchServiceDetails(this.props);
  }

  componentWillReceiveProps(nextProps: ServiceId) {
    this.fetchServiceDetails(nextProps);
  }

  fetchServiceDetails(props: ServiceId) {
    console.log('Fetching info of a service...');
    API.GetServiceDetail(props.namespace, props.service)
      .then(response => {
        let data = response['data'];
        this.setState({
          labels: data.labels,
          name: data.name,
          type: data.type,
          ports: data.ports,
          endpoints: data.endpoints,
          deployments: data.deployments,
          dependencies: data.dependencies,
          rules: data.route_rules,
          ip: data.ip
        });
      })
      .catch(error => {
        this.setState({
          error: true,
          errorMessage: 'Could not connect to server'
        });
        console.log(error);
      });
  }

  render() {
    return (
      <div>
        {this.state.error ? (
          <ToastNotificationList>
            <ToastNotification type="danger">
              <span>
                <strong>Error </strong>
                {this.state.errorMessage}
              </span>
            </ToastNotification>
          </ToastNotificationList>
        ) : null}
        <div className="container-fluid container-cards-pf">
          <Row className="row-cards-pf">
            <Col xs={12} sm={12} md={12} lg={12}>
              <ServiceInfoDescription
                name={this.state.name}
                labels={this.state.labels}
                ports={this.state.ports}
                type={this.state.type}
                ip={this.state.ip}
                endpoints={this.state.endpoints}
              />
            </Col>
          </Row>
          <Row className="row-cards-pf">
            <Col xs={12} sm={6} md={4} lg={4}>
              <ServiceInfoDeployments deployments={this.state.deployments} />
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <ServiceInfoRoutes dependencies={this.state.dependencies} />
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <ServiceInfoRules rules={this.state.rules} />
            </Col>
          </Row>
        </div>
      </div>
    );
  }
}

export default ServiceInfo;
