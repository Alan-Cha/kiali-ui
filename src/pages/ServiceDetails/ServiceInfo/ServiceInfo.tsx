import * as React from 'react';
import { Link } from 'react-router-dom';
import ServiceId from '../../../types/ServiceId';
import ServiceInfoBadge from './ServiceInfoBadge';
import ServiceInfoCard from './ServiceInfoCard';
import { Endpoints, Pod, Port, Source, Rule } from '../../../types/ServiceInfo';
import * as API from '../../../services/Api';
import { ToastNotification, ToastNotificationList, Col, Row } from 'patternfly-react';

type ServiceInfoState = {
  labels?: Map<string, string>;
  type: string;
  name: string;
  ip: string;
  ports?: Port[];
  endpoints?: Endpoints[];
  pods?: Pod[];
  rules?: Rule[];
  dependencies?: Source[];
  error: boolean;
  errorMessage: string;
};

class ServiceInfo extends React.Component<ServiceId, ServiceInfoState> {
  constructor(props: ServiceId) {
    super(props);
    this.state = {
      labels: new Map(),
      name: '',
      type: '',
      ip: '',
      ports: [],
      pods: [],
      rules: [],
      dependencies: [],
      error: false,
      errorMessage: ''
    };
  }

  componentWillMount() {
    this.fetchServiceDetails(this.props);
  }

  componentWillReceiveProps(nextProps: ServiceId) {
    this.fetchServiceDetails(nextProps);
  }

  fetchServiceDetails(props: ServiceId) {
    console.log('Fetching info of a service...');
    API.GetServiceDetail(props.namespace, props.service)
      .then(response => {
        console.log(response['data']);
        let data = response['data'];
        this.setState({
          labels: data.labels,
          name: data.name,
          type: data.type,
          ports: data.ports,
          endpoints: data.endpoints,
          pods: data.pods,
          dependencies: data.dependencies,
          rules: data.route_rules,
          ip: data.ip
        });
      })
      .catch(error => {
        this.setState({
          error: true,
          errorMessage: 'Could not connect to server'
        });
        console.log(error);
      });
  }

  render() {
    return (
      <div>
        {this.state.error ? (
          <ToastNotificationList>
            <ToastNotification type="danger">
              <span>
                <strong>Error </strong>
                {this.state.errorMessage}
              </span>
            </ToastNotification>
          </ToastNotificationList>
        ) : null}
        <div className="container-fluid container-cards-pf">
          <Row className="row-cards-pf">
            <Col>
              <ServiceInfoCard
                iconType="pf"
                iconName="service"
                title={this.state.name}
                items={
                  <Row>
                    <Col xs={12} sm={6} md={2} lg={2}>
                      <div className="progress-description">
                        <strong>Labels</strong>
                      </div>
                      {Object.keys(this.state.labels || new Map()).map((key, i) => (
                        <ServiceInfoBadge
                          key={'label_' + i}
                          scale={0.8}
                          style="plastic"
                          color="#0088ce"
                          leftText={key}
                          rightText={this.state.labels ? this.state.labels[key] : ''}
                        />
                      ))}
                      <div>
                        <strong>Type</strong> {this.state.type ? this.state.type : ''}
                      </div>
                      <div>
                        <strong> Ip</strong> {this.state.ip ? this.state.ip : ''}
                      </div>
                    </Col>
                    <Col xs={12} sm={6} md={4} lg={4}>
                      <div className="progress-description">
                        <strong>Ports</strong>
                      </div>
                      <ul style={{ listStyleType: 'none' }}>
                      {(this.state.ports || []).map((port, i) => (
                        <li key={'port_' + i}>
                          {port.protocol} {port.name} ({port.port})
                        </li>
                      ))}
                      </ul>
                    </Col>
                    <Col xs={12} sm={6} md={6} lg={6}>
                      <div className="progress-description">
                        <strong>Endpoints</strong>
                      </div>
                      {(this.state.endpoints || []).map((endpoint, i) => (
                        <Row key={'endpoint_' + i}>
                          <Col xs={12} sm={12} md={12} lg={12}>
                            <ul style={{ listStyleType: 'none' }}>
                              {(endpoint.addresses || []).map((address, u) => (
                                <li key={'endpoint_' + i + '_address_' + u}>
                                  <strong>{address.ip} </strong>: {address.name}
                                </li>
                              ))}
                            </ul>
                          </Col>
                        </Row>
                      ))}
                    </Col>
                  </Row>
                }
              />
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <ServiceInfoCard
                iconType="fa"
                iconName="cube"
                title="Pods"
                items={(this.state.pods || []).map((pod, u) => (
                  <div key={'pods_' + u}>
                    <div>
                      <strong>Pod</strong>: {pod['name']}
                    </div>
                    <ul style={{ listStyleType: 'none' }}>
                      {Object.keys(pod.labels || new Map()).map((key, i) => (
                        <li>
                          <ServiceInfoBadge
                            key={'pod_labels_badge_' + i}
                            scale={0.8}
                            style="plastic"
                            color="green"
                            leftText={key}
                            rightText={pod.labels ? pod.labels[key] : ''}
                          />
                        </li>
                      ))}
                    </ul>
                  </div>
                ))}
              />
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <ServiceInfoCard
                iconType="pf"
                iconName="route"
                title="Source Services"
                items={Object.keys(this.state.dependencies || new Map()).map((key, u) => (
                  <div key={'dependencies_' + u}>
                    <div className="progress-description">
                      <strong>To: </strong> {key}
                    </div>
                    <ul style={{ listStyleType: 'none' }}>
                      {(this.state.dependencies ? this.state.dependencies[key] : []).map((dependency, i) => {
                        let nVersion = dependency.indexOf('/');
                        let nNamespace = dependency.indexOf('.');
                        let servicename = dependency.substring(0, nNamespace);
                        let namespace = dependency.substring(nNamespace + 1, nVersion);
                        if (servicename.length > 0 && namespace.length > 0) {
                          let to = '/namespaces/' + namespace + '/services/' + servicename;
                          return (
                            <Link key={to} to={to}>
                              <li key={'dependencies_' + u + '_dependency_' + i}>{dependency}</li>
                            </Link>
                          );
                        } else {
                          return <li key={'dependencies_' + u + '_dependency_' + i}>{dependency}</li>;
                        }
                      })}
                    </ul>
                  </div>
                ))}
              />
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <ServiceInfoCard
                iconType="pf"
                iconName="settings"
                title="Istio Route Rules"
                items={(this.state.rules || []).map((rule, i) => (
                  <div key={'rule' + i}>
                    <div>
                      <strong>Name</strong> : {rule.name}
                    </div>
                    <div>
                      <strong>Precendence</strong> : {rule.precedence}
                    </div>
                    <div>
                      <strong>Route</strong>:
                      <ul style={{ listStyleType: 'none' }}>
                        {(rule.route || []).map((label, u) =>
                          Object.keys(label.labels || new Map()).map((key, n) => (
                            <li key={'rule_' + i + '_label_' + u + '_n_' + n}>
                              <ServiceInfoBadge
                                key={'rule_labels_badge_' + i}
                                scale={0.8}
                                style="plastic"
                                color="green"
                                leftText={key}
                                rightText={label.labels ? label.labels[key] : ''}
                              />
                            </li>
                          ))
                        )}
                      </ul>
                    </div>
                    <div>
                      {!rule.match ? null : (
                        <div>
                          <strong>Match</strong>:
                          <textarea className="form-control textarea-resize" readOnly={true}>
                            {JSON.stringify(rule.match, null, 2)}
                          </textarea>
                        </div>
                      )}
                    </div>
                    <hr />
                  </div>
                ))}
              />
            </Col>
          </Row>
        </div>
      </div>
    );
  }
}

export default ServiceInfo;
