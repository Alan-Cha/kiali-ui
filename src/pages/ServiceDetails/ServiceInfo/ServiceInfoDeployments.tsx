import * as React from 'react';
import Badge from '../../../components/Badge/Badge';
import { Deployment } from '../../../types/ServiceInfo';
import { Col, Row, Icon } from 'patternfly-react';
import PfInfoCard from '../../../components/Pf/PfInfoCard';

interface ServiceInfoDeploymentsProps {
  deployments?: Deployment[];
}

class ServiceInfoDeployments extends React.Component<ServiceInfoDeploymentsProps> {
  constructor(props: ServiceInfoDeploymentsProps) {
    super(props);
  }

  render() {
    return (
      <PfInfoCard
        iconType="fa"
        iconName="cube"
        title="Deployments"
        items={(this.props.deployments || []).map((deployment, u) => (
          <Row key={'deployments_' + u}>
            <Col xs={12}>
              <div>
                <p>
                  <strong>{deployment.name}</strong>
                </p>
                <div key="labels">
                  {Object.keys(deployment.labels || new Map()).map((key, i) => (
                    <Badge
                      key={'deployment_' + i}
                      scale={0.8}
                      style="plastic"
                      color="green"
                      leftText={key}
                      rightText={deployment.labels ? deployment.labels[key] : ''}
                    />
                  ))}
                </div>
                <div>
                  <strong>Pod status: </strong> {deployment.available_replicas} / {deployment.replicas}{' '}
                  <Icon
                    type="pf"
                    name={deployment.available_replicas < deployment.replicas ? 'warning-triangle-o' : 'ok'}
                  />
                </div>
                {deployment.autoscaler.name !== '' && (
                  <div>
                    <strong>Autoscaler: </strong>
                    from {deployment.autoscaler.min_replicas} to {deployment.autoscaler.max_replicas} pods ({
                      deployment.autoscaler.target_cpu_utilization_percentage
                    }% CPU)
                  </div>
                )}
              </div>
              <hr />
            </Col>
          </Row>
        ))}
      />
    );
  }
}

export default ServiceInfoDeployments;
