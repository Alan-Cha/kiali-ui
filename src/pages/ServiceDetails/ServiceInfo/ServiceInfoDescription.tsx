import * as React from 'react';
import ServiceInfoBadge from './ServiceInfoBadge';
import ServiceInfoCard from './ServiceInfoCard';
import { Col, Row } from 'patternfly-react';
import { Endpoints, Port } from '../../../types/ServiceInfo';

interface ServiceInfoDescriptionProps {
  name?: string;
  labels?: Map<string, string>;
  type?: string;
  ip?: any;
  ports?: Port[];
  endpoints?: Endpoints[];
}

class ServiceInfoDescription extends React.Component<ServiceInfoDescriptionProps> {
  constructor(props: ServiceInfoDescriptionProps) {
    super(props);
  }

  render() {
    return (
      <ServiceInfoCard
        iconType="pf"
        iconName="service"
        title={this.props.name}
        items={
          <Row>
            <Col xs={12} sm={6} md={2} lg={2}>
              <div className="progress-description">
                <strong>Labels</strong>
              </div>
              {Object.keys(this.props.labels || new Map()).map((key, i) => (
                <div key={'label_' + i}>
                  <ServiceInfoBadge
                    scale={0.8}
                    style="plastic"
                    color="#0088ce"
                    leftText={key}
                    rightText={this.props.labels ? this.props.labels[key] : ''}
                  />
                </div>
              ))}
              <div>
                <strong>Type</strong> {this.props.type ? this.props.type : ''}
              </div>
              <div>
                <strong> Ip</strong> {this.props.ip ? this.props.ip : ''}
              </div>
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <div className="progress-description">
                <strong>Ports</strong>
              </div>
              <ul style={{ listStyleType: 'none' }}>
                {(this.props.ports || []).map((port, i) => (
                  <li key={'port_' + i}>
                    {port.protocol} {port.name} ({port.port})
                  </li>
                ))}
              </ul>
            </Col>
            <Col xs={12} sm={6} md={6} lg={6}>
              <div className="progress-description">
                <strong>Endpoints</strong>
              </div>
              {(this.props.endpoints || []).map((endpoint, i) => (
                <Row key={'endpoint_' + i}>
                  <Col xs={12} sm={12} md={12} lg={12}>
                    <ul style={{ listStyleType: 'none' }}>
                      {(endpoint.addresses || []).map((address, u) => (
                        <li key={'endpoint_' + i + '_address_' + u}>
                          <strong>{address.ip} </strong>: {address.name}
                        </li>
                      ))}
                    </ul>
                  </Col>
                </Row>
              ))}
            </Col>
          </Row>
        }
      />
    );
  }
}

export default ServiceInfoDescription;
