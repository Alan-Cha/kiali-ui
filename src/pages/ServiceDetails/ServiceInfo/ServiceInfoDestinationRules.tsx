import * as React from 'react';
import { DestinationRule } from '../../../types/ServiceInfo';
import { Col, Row } from 'patternfly-react';
import LocalTime from '../../../components/Time/LocalTime';
import DetailObject from '../../../components/Details/DetailObject';

interface ServiceInfoDestinationRulesProps {
  destinationRules?: DestinationRule[];
}

class ServiceInfoDestinationRules extends React.Component<ServiceInfoDestinationRulesProps> {
  constructor(props: ServiceInfoDestinationRulesProps) {
    super(props);
  }

  render() {
    return (
      <div className="card-pf">
        <Row className="row-cards-pf">
          <Col xs={12} sm={12} md={12} lg={12}>
            {(this.props.destinationRules || []).map((destinationRule, i) => (
              <div className="card-pf-body" key={'virtualService' + i}>
                <div>
                  <strong>Name</strong>: {destinationRule.name}
                </div>
                <div>
                  <strong>Created at</strong>: <LocalTime time={destinationRule.created_at} />
                </div>
                <div>
                  <strong>Resource Version</strong>: {destinationRule.resource_version}
                </div>
                {destinationRule.trafficPolicy ? (
                  <DetailObject name="Traffic Policy" detail={destinationRule.trafficPolicy} />
                ) : (
                  undefined
                )}
                {destinationRule.subsets && destinationRule.subsets.length > 0 ? (
                  <DetailObject name="Subsets" detail={destinationRule.subsets} labels={['labels']} />
                ) : (
                  undefined
                )}
                <hr />
              </div>
            ))}
          </Col>
        </Row>
      </div>
    );
  }
}

export default ServiceInfoDestinationRules;
