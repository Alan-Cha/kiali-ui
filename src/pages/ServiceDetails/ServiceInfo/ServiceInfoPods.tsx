import * as React from 'react';
import ServiceInfoBadge from './ServiceInfoBadge';
import ServiceInfoCard from './ServiceInfoCard';
import { Pod } from '../../../types/ServiceInfo';

interface ServiceInfoPodsProps {
  pods?: Pod[];
}

class ServiceInfoPods extends React.Component<ServiceInfoPodsProps> {
  constructor(props: ServiceInfoPodsProps) {
    super(props);
  }

  render() {
    return (
      <ServiceInfoCard
        iconType="fa"
        iconName="cube"
        title="Pods"
        items={(this.props.pods || []).map((pod, u) => (
          <div key={'pods_' + u}>
            <div>
              <strong>Pod</strong>: {pod['name']}
            </div>
            <ul style={{ listStyleType: 'none' }}>
              {Object.keys(pod.labels || new Map()).map((key, i) => (
                <li key={'pod_labels_badge_' + i}>
                  <ServiceInfoBadge
                    scale={0.8}
                    style="plastic"
                    color="green"
                    leftText={key}
                    rightText={pod.labels ? pod.labels[key] : ''}
                  />
                </li>
              ))}
            </ul>
          </div>
        ))}
      />
    );
  }
}

export default ServiceInfoPods;
