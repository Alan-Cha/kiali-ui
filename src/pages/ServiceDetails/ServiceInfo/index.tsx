import { default as ServiceInfoDeployments } from './ServiceInfoDeployments';
import { default as ServiceInfoDescription } from './ServiceInfoDescription';
import { default as ServiceInfoRoutes } from './ServiceInfoRoutes';
import { default as ServiceInfoRules } from './ServiceInfoRules';
import { default as ServiceInfoBadge } from './ServiceInfoBadge';

export { ServiceInfoDeployments, ServiceInfoDescription, ServiceInfoRoutes, ServiceInfoRules, ServiceInfoBadge };
