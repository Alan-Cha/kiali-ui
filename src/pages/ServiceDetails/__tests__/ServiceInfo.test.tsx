import * as React from 'react';
import { shallow } from 'enzyme';
import ServiceInfo from '../ServiceInfo';

jest.mock('../../../services/Api');

describe('#ServiceInfo render correctly with data', () => {
  it('should render serviceInfo', () => {
    const wrapper = shallow(<ServiceInfo namespace="istio-system" service="reviews" />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('ServiceInfoDescription').length === 1).toBeTruthy();
    expect(wrapper.find('ServiceInfoPods').length === 1).toBeTruthy();
    expect(wrapper.find('ServiceInfoRoutes').length === 1).toBeTruthy();
    expect(wrapper.find('ServiceInfoRules').length === 1).toBeTruthy();
  });
});
