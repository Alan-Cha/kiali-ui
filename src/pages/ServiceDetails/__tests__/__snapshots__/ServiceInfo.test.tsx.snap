// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ServiceInfo render correctly with data should render serviceInfo 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ServiceInfo
    namespace="istio-system"
    service="reviews"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <div
          className="container-fluid container-cards-pf"
>
          <Row
                    bsClass="row"
                    className="row-cards-pf"
                    componentClass="div"
          >
                    <Col
                              bsClass="col"
                              componentClass="div"
                    >
                              <ServiceInfoDescription
                                        endpoints={undefined}
                                        ip=""
                                        labels={Map {}}
                                        name=""
                                        ports={Array []}
                                        type=""
                              />
                    </Col>
                    <Col
                              bsClass="col"
                              componentClass="div"
                              lg={4}
                              md={4}
                              sm={6}
                              xs={12}
                    >
                              <ServiceInfoPods
                                        pods={Array []}
                              />
                    </Col>
                    <Col
                              bsClass="col"
                              componentClass="div"
                              lg={4}
                              md={4}
                              sm={6}
                              xs={12}
                    >
                              <ServiceInfoRoutes
                                        dependencies={Map {}}
                              />
                    </Col>
                    <Col
                              bsClass="col"
                              componentClass="div"
                              lg={4}
                              md={4}
                              sm={6}
                              xs={12}
                    >
                              <ServiceInfoRules
                                        rules={Array []}
                              />
                    </Col>
          </Row>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Row
            bsClass="row"
            className="row-cards-pf"
            componentClass="div"
>
            <Col
                        bsClass="col"
                        componentClass="div"
            >
                        <ServiceInfoDescription
                                    endpoints={undefined}
                                    ip=""
                                    labels={Map {}}
                                    name=""
                                    ports={Array []}
                                    type=""
                        />
            </Col>
            <Col
                        bsClass="col"
                        componentClass="div"
                        lg={4}
                        md={4}
                        sm={6}
                        xs={12}
            >
                        <ServiceInfoPods
                                    pods={Array []}
                        />
            </Col>
            <Col
                        bsClass="col"
                        componentClass="div"
                        lg={4}
                        md={4}
                        sm={6}
                        xs={12}
            >
                        <ServiceInfoRoutes
                                    dependencies={Map {}}
                        />
            </Col>
            <Col
                        bsClass="col"
                        componentClass="div"
                        lg={4}
                        md={4}
                        sm={6}
                        xs={12}
            >
                        <ServiceInfoRules
                                    rules={Array []}
                        />
            </Col>
</Row>,
          "className": "container-fluid container-cards-pf",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": Array [
              <Col
                bsClass="col"
                componentClass="div"
>
                <ServiceInfoDescription
                                endpoints={undefined}
                                ip=""
                                labels={Map {}}
                                name=""
                                ports={Array []}
                                type=""
                />
</Col>,
              <Col
                bsClass="col"
                componentClass="div"
                lg={4}
                md={4}
                sm={6}
                xs={12}
>
                <ServiceInfoPods
                                pods={Array []}
                />
</Col>,
              <Col
                bsClass="col"
                componentClass="div"
                lg={4}
                md={4}
                sm={6}
                xs={12}
>
                <ServiceInfoRoutes
                                dependencies={Map {}}
                />
</Col>,
              <Col
                bsClass="col"
                componentClass="div"
                lg={4}
                md={4}
                sm={6}
                xs={12}
>
                <ServiceInfoRules
                                rules={Array []}
                />
</Col>,
            ],
            "className": "row-cards-pf",
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <ServiceInfoDescription
                  endpoints={undefined}
                  ip=""
                  labels={Map {}}
                  name=""
                  ports={Array []}
                  type=""
/>,
                "componentClass": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "endpoints": undefined,
                  "ip": "",
                  "labels": Map {},
                  "name": "",
                  "ports": Array [],
                  "type": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <ServiceInfoPods
                  pods={Array []}
/>,
                "componentClass": "div",
                "lg": 4,
                "md": 4,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "pods": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <ServiceInfoRoutes
                  dependencies={Map {}}
/>,
                "componentClass": "div",
                "lg": 4,
                "md": 4,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "dependencies": Map {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <ServiceInfoRules
                  rules={Array []}
/>,
                "componentClass": "div",
                "lg": 4,
                "md": 4,
                "sm": 6,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "rules": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="container-fluid container-cards-pf"
>
            <Row
                        bsClass="row"
                        className="row-cards-pf"
                        componentClass="div"
            >
                        <Col
                                    bsClass="col"
                                    componentClass="div"
                        >
                                    <ServiceInfoDescription
                                                endpoints={undefined}
                                                ip=""
                                                labels={Map {}}
                                                name=""
                                                ports={Array []}
                                                type=""
                                    />
                        </Col>
                        <Col
                                    bsClass="col"
                                    componentClass="div"
                                    lg={4}
                                    md={4}
                                    sm={6}
                                    xs={12}
                        >
                                    <ServiceInfoPods
                                                pods={Array []}
                                    />
                        </Col>
                        <Col
                                    bsClass="col"
                                    componentClass="div"
                                    lg={4}
                                    md={4}
                                    sm={6}
                                    xs={12}
                        >
                                    <ServiceInfoRoutes
                                                dependencies={Map {}}
                                    />
                        </Col>
                        <Col
                                    bsClass="col"
                                    componentClass="div"
                                    lg={4}
                                    md={4}
                                    sm={6}
                                    xs={12}
                        >
                                    <ServiceInfoRules
                                                rules={Array []}
                                    />
                        </Col>
            </Row>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Row
              bsClass="row"
              className="row-cards-pf"
              componentClass="div"
>
              <Col
                            bsClass="col"
                            componentClass="div"
              >
                            <ServiceInfoDescription
                                          endpoints={undefined}
                                          ip=""
                                          labels={Map {}}
                                          name=""
                                          ports={Array []}
                                          type=""
                            />
              </Col>
              <Col
                            bsClass="col"
                            componentClass="div"
                            lg={4}
                            md={4}
                            sm={6}
                            xs={12}
              >
                            <ServiceInfoPods
                                          pods={Array []}
                            />
              </Col>
              <Col
                            bsClass="col"
                            componentClass="div"
                            lg={4}
                            md={4}
                            sm={6}
                            xs={12}
              >
                            <ServiceInfoRoutes
                                          dependencies={Map {}}
                            />
              </Col>
              <Col
                            bsClass="col"
                            componentClass="div"
                            lg={4}
                            md={4}
                            sm={6}
                            xs={12}
              >
                            <ServiceInfoRules
                                          rules={Array []}
                            />
              </Col>
</Row>,
            "className": "container-fluid container-cards-pf",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "row",
              "children": Array [
                <Col
                  bsClass="col"
                  componentClass="div"
>
                  <ServiceInfoDescription
                                    endpoints={undefined}
                                    ip=""
                                    labels={Map {}}
                                    name=""
                                    ports={Array []}
                                    type=""
                  />
</Col>,
                <Col
                  bsClass="col"
                  componentClass="div"
                  lg={4}
                  md={4}
                  sm={6}
                  xs={12}
>
                  <ServiceInfoPods
                                    pods={Array []}
                  />
</Col>,
                <Col
                  bsClass="col"
                  componentClass="div"
                  lg={4}
                  md={4}
                  sm={6}
                  xs={12}
>
                  <ServiceInfoRoutes
                                    dependencies={Map {}}
                  />
</Col>,
                <Col
                  bsClass="col"
                  componentClass="div"
                  lg={4}
                  md={4}
                  sm={6}
                  xs={12}
>
                  <ServiceInfoRules
                                    rules={Array []}
                  />
</Col>,
              ],
              "className": "row-cards-pf",
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "col",
                  "children": <ServiceInfoDescription
                    endpoints={undefined}
                    ip=""
                    labels={Map {}}
                    name=""
                    ports={Array []}
                    type=""
/>,
                  "componentClass": "div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "endpoints": undefined,
                    "ip": "",
                    "labels": Map {},
                    "name": "",
                    "ports": Array [],
                    "type": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "col",
                  "children": <ServiceInfoPods
                    pods={Array []}
/>,
                  "componentClass": "div",
                  "lg": 4,
                  "md": 4,
                  "sm": 6,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "pods": Array [],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "col",
                  "children": <ServiceInfoRoutes
                    dependencies={Map {}}
/>,
                  "componentClass": "div",
                  "lg": 4,
                  "md": 4,
                  "sm": 6,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "dependencies": Map {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "col",
                  "children": <ServiceInfoRules
                    rules={Array []}
/>,
                  "componentClass": "div",
                  "lg": 4,
                  "md": 4,
                  "sm": 6,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "rules": Array [],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
